source/transfer_list.rst

```diff
@@ -657,5 +660,120 @@ such that the last ACPI table in this entry ends at offset
      - hdr_size
      - One or more ACPI tables.
 
+.. _optee_pagable_part_entry:
+
+OP-TEE pagable part entry layout (XFERLIST_OPTEE_PAGABLE_PART)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This entry type holds the address of OP-TEE pagable part.
+This address (of type 'uint64_t') is used when OPTEED(OP-TEE Dispatcher)
+is the Secure Payload Dispatcher, indicating where to load the pagable image of
+OP-TEE OS.
+
+.. _tab_optee_pagable_part:
+.. list-table:: OP-TEE pagable part type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **0x20**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size of OP-TEE pagable part in bytes must be set to **8**.
+
+   * - pagable_part
+     - data_size
+     - hdr_size
+     - Pagable part address for OPTEED
+
+.. _tos_fw_config_entry:
+
+TOS FW config entry layout (XFERLIST_TOS_FW_CONFIG)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This entry type holds the SPMC(Secure Partition Manager Core) manifest image.
+It is used when OP-TEE represents SPMC at S-EL1(Secure Exception Level 1) and
+SPMD(Secure Partition Manager Dispatcher) is the Secure Payload Dispatcher.
+
+.. _tab_tos_fw_config:
+.. list-table:: TOS FW config type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **0x21**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size of TOS FW config in bytes.
+
+   * - tos_fw_config
+     - data_size
+     - hdr_size
+     - TOS FW config base address
+
+.. _opteed_aarch_entry:
+
+OPTEED(OP-TEE Dispatcher) AARCH entry layout (XFERLIST_OPTEED_AARCH)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This entry type holds the AARCH mode for OPTEED.
+This AARCH mode (of type 'uint32_t', **0** for AARCH64 and **1** for AARCH32)
+is used when OPTEED is the Secure Payload Dispatcher.
+
+.. _tab_opteed_aarch:
+.. list-table:: OPTEED AARCH type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **0x22**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size of OPTEED AARCH mode in bytes must be set to **4**.
+
+   * - opteed_aarch
```

Comment:
Why do you need a separate TE for this? Can't this be merged with XFERLIST_OPTEE_PAGABLE_PART and use a more generic name to represent the data that the OPTEED passes to OP-TEE.

Reply:
Generally this TE is used in between BL2 and opteed within TF-A scope only, but will not be in OP-TEE, which is not the same case comparing with  `XFERLIST_OPTEE_PAGABLE_PART`.